File: app/src/main/java/com/worldventures/dreamtrips/modules/friends/service/command/AcceptAllFriendRequestsCommand.java
package com.worldventures.dreamtrips.modules.friends.service.command;

import com.worldventures.dreamtrips.R;
import com.worldventures.dreamtrips.api.friends.AcceptAllFriendRequestsHttpAction;
import com.worldventures.dreamtrips.core.api.action.CommandWithError;
import com.worldventures.dreamtrips.core.janet.dagger.InjectableAction;

import javax.inject.Inject;

import io.techery.janet.Janet;
import io.techery.janet.command.annotations.CommandAction;
import rx.schedulers.Schedulers;

@CommandAction
public class AcceptAllFriendRequestsCommand extends CommandWithError<Void> implements InjectableAction {

   private String circleId;

   @Inject Janet janet;

   public AcceptAllFriendRequestsCommand(String circleId) {
      this.circleId = circleId;
   }

   @Override
   protected void run(CommandCallback<Void> callback) {
      janet.createPipe(AcceptAllFriendRequestsHttpAction.class, Schedulers.io())
            .createObservableResult(new AcceptAllFriendRequestsHttpAction(circleId))
            .subscribe(action -> callback.onSuccess(null), callback::onFail);
   }

   @Override
   public int getFallbackErrorMessage() {
      return R.string.error_fail_to_accept_friend_request;
   }

}

File: app/src/main/java/com/worldventures/dreamtrips/social/service/users/request/command/AcceptAllFriendRequestsCommand.kt
package com.worldventures.dreamtrips.social.service.users.request.command

import com.worldventures.core.janet.CommandWithError
import com.worldventures.dreamtrips.R
import com.worldventures.dreamtrips.api.friends.AcceptAllFriendRequestsHttpAction
import com.worldventures.janet.injection.InjectableAction
import io.techery.janet.Janet
import io.techery.janet.command.annotations.CommandAction
import rx.schedulers.Schedulers
import javax.inject.Inject

@CommandAction
class AcceptAllFriendRequestsCommand(val circleId: String) : CommandWithError<Void>(), InjectableAction {

   @Inject lateinit var janet: Janet

   @Throws(Throwable::class)
   override fun run(callback: CommandCallback<Void>) {
      janet.createPipe(AcceptAllFriendRequestsHttpAction::class.java, Schedulers.io())
            .createObservableResult(AcceptAllFriendRequestsHttpAction(circleId))
            .subscribe({ callback.onSuccess(null) }, callback::onFail)
   }

   override fun getFallbackErrorMessage() = R.string.error_fail_to_accept_friend_request

}
